src/os/bluestore/BlueStore.h

```diff
@@ -513,8 +513,9 @@ class BlueStore : public ObjectStore,
       ++nref;
     }
     void put() {
-      if (--nref == 0)
-	delete this;
+      if (--nref == 0) {
+	//delete this;
+      }
```

Comment:
According to our today's conversation: `boost::object_pool` [guarantees](http://www.boost.org/doc/libs/1_60_0/libs/pool/doc/html/boost/object_pool.html) the memory will be in a consistent state in the meaning of the encapsulated instance's construction and destruction:

> Class object_pool is a template class that can be used for fast and efficient memory allocation of objects. **It also provides automatic destruction of non-deallocated objects.**
>
> When the object pool is destroyed, then the destructor for type T is called for each allocated T that has not yet been deallocated.

However, the plan is to destruct `Blob` that isn't necessary anymore much earlier -- together with a *last* corresponding `Extent`.